hyperspy/drawing/spectrum.py

```diff
@@ -331,7 +331,16 @@ def update(self, force_replot=False):
             ydata = self.data_function(axes_manager=self.axes_manager).real
         else:
             ydata = self.data_function(axes_manager=self.axes_manager).imag
-        self.line.set_ydata(ydata)
+
+        new_axis = self.axes_manager.signal_axes[0].axis
```

Comment:
This line currently breaks navigation, as the axis plotted isn't always `signal_axes[0]`, e.g. when the Spectrum plot is for the navigation plot.